extensions/2.0/Vendor/EXT_feature_metadata/README.md

```diff
@@ -0,0 +1,1691 @@
+- [Revision History](#revision-history)
+
+## Overview
+
+A **feature** is an entity that has both geometry and metadata. In Geographic Information Systems (GIS) a feature is an entity such as a point, polyline, or polygon that represents some element on a map. In another domain like CAD/BIM a feature might be a component of a design model. A feature could also be a 3D building in a city, a tree in a forest, a sample point in a weather model, or a patch of texels on a 3D model.
+
+This extension allows batching features for efficient streaming to a client for rendering and interaction. Efficiency comes from transferring multiple features in the same glTF and rendering them in the least number of draw calls necessary.
+
+Feature IDs enable individual features to be identified and updated at runtime. For example, a selected feature could be shown/hidden, or highlighted a different color. Feature IDs may be assigned on a per-vertex, per-texel, or per-instance basis.
+
+Feature IDs may be used to access metadata, such as passing a building's ID to get its address. Feature metadata is stored in a compact binary tabular format described in the [Cesium 3D Metadata Specification](https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/specification/Metadata).
+
+![Building Example](figures/feature-table-buildings.png)
+
+In the image above, a glTF consists of two houses batched together into a single primitive. A feature ID attribute on the primitive indicates that all of the vertices making up the first house have a feature ID of 0, while all vertices making up the second house have the feature ID 1. The feature ID is then used to access the building's metadata from the feature table.
+
+Feature metadata may also be stored directly in textures. This is especially useful when texture mapping high frequency data, such as material properties, to less detailed 3D surfaces. Feature textures enable new styling and analytics capabilities, and complement glTF PBR textures.
+
+See [Examples](#examples) for a full list of use cases for this extension.
+
+## Feature Identification
+
+Features in a glTF primitive are identified in three ways:
+
+* Per-vertex using a vertex attribute
+* Per-texel using a glTF texture
+* Per-instance using an instance attribute with the [`EXT_mesh_gpu_instancing` extension](../../EXT_mesh_gpu_instancing)
+
+<img src="figures/metadata-access.png"  alt="Metadata Access" width="600">
+
+### Feature ID Vertex Attributes
+
+#### Feature ID Accessors
+
+The most straightforward method for defining feature IDs is to store them in a glTF vertex attribute. Feature ID attributes must follow the naming convention `_FEATURE_ID_X` where `X` is a non-negative integer. The first feature ID attribute is `_FEATURE_ID_0`, the second `_FEATURE_ID_1`, and so on.
+
+Feature IDs must be whole numbers in the range `[0, count - 1]` (inclusive), where `count` is the total number of features in the feature table.
```

Comment:
I expect this is more of a change in structure than is desirable at this point, but I'd like to mention it anyway – the per-vertex identification of features is useful for _lookups_ but may be limiting in terms of the interactivity you can support operating on those features. For example, displaying an outline around a feature, or moving/scaling/rotating it, would be quite complex.

A natural way to support those use cases would be [WEBGL_multi_draw](https://developer.mozilla.org/en-US/docs/Web/API/WEBGL_multi_draw), which for best results would require: (1) a feature's vertices should be in contiguous memory, (2) offset and vertex/index count for each feature should be known.

Maybe this doesn't require structural change, but instead requires additional requirements (contiguous memory) and metadata (counts and offsets) which could be added in a future extension.

Reply:
We've considered the count/offset approach as well. The main downsides is that it's not possible to support multiple feature layers - the current approach allows you to add `FEATURE_ID_0` and `FEATURE_ID_1` to the same primitive. This is an important use case for supporting multiple label sets on the same mesh.

Reply:
Ok, yes, contiguous memory is not compatible with that use case, and it sounds like multiple feature layers are (a) more important, and (b) harder to defer to another extension.

I don't see any reason that the requirement of contiguous memory, and the addition of counts/offsets, couldn't be added in a future extension to support more interactivity if needed, so it seems fine leave to these out in `EXT_feature_metadata`. 👍